### PR TITLE
Add Generic.Arrays.ArrayIndent sniff

### PR DESCRIPTION
**Pull request by deeky666**
Seems this is not part of PSR-2 -.-

### CODE REVIEW

File: lib/Doctrine/ruleset.xml [hunk 1]
```diff
@@ -8,6 +8,8 @@
         <exclude name="PSR2.Namespaces.UseDeclaration.SpaceAfterLastUse"/>
     </rule>
 
+    <!-- Indent array keys with 4 spaces -->
```
Comment (carusogabriel) [inline lib/Doctrine/ruleset.xml]: Maybe "Force array formatting"

File: tests/input/array_indentation.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    4,
+     5,
+];
+
+$multiArray = [
+    4 => [
+       7,
+        8,
+         9,
+    ],
```
Comment (carusogabriel) [inline tests/input/array_indentation.php]: Please update the tests with a example with `string`s in the keys, like:
```php
$foo = [
'foo'   =>  [
  'bar' => 2,
1 => 'baz',
];
];
```

Comment (lcobucci) [reply tests/input/array_indentation.php]: Maybe this is also an interesting scenario:

```php
$items->append($one)
      ->append(
            [
0,
 1,
  2,
   ...,
              14,
               15,
                16,
            ]
      );
```

And I'd expect to have something like:


```php
$items->append($one)
      ->append(
            [
                0,
                1,
                2,
                ...,
                14,
                15,
                16,
            ]
      );
```

File: lib/Doctrine/ruleset.xml [hunk 3]
```diff
@@ -17,6 +19,7 @@
         <!-- But allow empty catch -->
         <exclude name="Generic.CodeAnalysis.EmptyStatement.DetectedCatch"/>
     </rule>
+    <rule ref="Generic.CodeAnalysis.ForLoopShouldBeWhileLoop"/>
```
Comment (carusogabriel) [inline lib/Doctrine/ruleset.xml]: What is this rule?

Comment (deeky666) [reply lib/Doctrine/ruleset.xml]: oh sry, that was something else I accidently commited xD. will fix it

File: tests/input/array_indentation.php (hunk 2 above)
Comment (lcobucci) [reply tests/input/array_indentation.php]: @deeky666 that makes sense =)

**Pull request merged**